aws/README.md

```diff
@@ -1,3 +1,189 @@
+
+#### Install Terraform
+
+Terraform is used to automate the manipulation of cloud infrastructure. Its [Terraform installation instructions](https://www.terraform.io/intro/getting-started/install.html) are also available online.
+
+#### Install aws-iam-authenticator
+
+Amazon EKS uses IAM to provide authentication to your Kubernetes cluster through the AWS IAM authenticator for Kubernetes. Follow the instructions to install [aws-iam-authenticator installation instructions](https://docs.aws.amazon.com/eks/latest/userguide/install-aws-iam-authenticator.html).
+
+#### Install kubectl
+Kubernetes uses a command line utility called kubectl for communicating with the cluster API server. The kubectl binary is available in many operating system package managers, and this option is often much easier than a manual download and install process. Follow the instructions to install [kubectl installation instructions](https://docs.aws.amazon.com/eks/latest/userguide/install-kubectl.html).
+
+### Configure AWS CLI
+Run `aws configure` command to configure your AWS cli, This Terraform module AWS CLI v2, if you have older versions of AWS CLI use the following instructions to uninstall older versions.
+
+* [uninstall AWS cli v1](https://docs.aws.amazon.com/cli/latest/userguide/install-linux.html)
+* [Install AWS cli v2](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html)
+
+```console
+aws configure
+```
+Enter `access key`, `secret access key` default `region`.
+
+## Test this project locally
+
+Export the following terraform environment variables(TFVARS) for terraform to create the resources. 
+```console
+
+export TF_VAR_environment=<ENVIRONMENT_REPLACEME>
+ex:- export TF_VAR_environment=dev
+
+export TF_VAR_name=<CLUSTERNAME_REPLACEME>
+ex:- export TF_VAR_name=k8ssandra
+
+export TF_VAR_region=<REGION_REPLACEME>
+ex:- export TF_VAR_region=us-east-1
+
+```
+
+Important: Initialize the terraform modules delete the backend file for local testing.
+
+```console
+cd env/
+terraform init
+````
+
+After the terraform initialization is successful, create your workspace and by using the following command
+
+```console
+terraform workspace new <WORKSPACENAME_REPLACEME>
+```
+
+or select the workspace if there are any existing workspaces
+
+```console
+terraform workspace select <WORKSPACENAME_REPLACEME>
+```
+
+Run the following commands to apply changes to your infrastructure.
+
+```console
+terraform plan
+terraform apply
+```
+
+To destroy the resource, use the following instructions:
+
+Create your workspace with the environment name, it is the recommended way of working with the Terraform workspaces among your teams. Select the workspace where resources need to be destroyed.
+
+It is important to export the same values when destroying the resources on a workspace. Make sure you exported the right environment variables (TF_VAR).
+
+```console
+terraform workspace select <WORKSPACENAME_REPLACEME>
```

Comment:
Are we doing this out of the Makefile?

Reply:
yes, If we want to create the resources locally.